src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -235,42 +235,58 @@ string CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::print_extern_call(const Call *op) {
     return rhs.str();
 }
 
+string CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::print_array_access(const string& name,
+                                                                const Type& type,
+                                                                const string& id_index) {
+    ostringstream rhs;
+    bool type_cast_needed = !(allocations.contains(name) &&
+                              allocations.get(name).type == type);
+
+    if (type_cast_needed) {
+        rhs << "((" << get_memory_space(name) << " "
+            << print_type(type) << " *)"
+            << print_name(name)
+            << ")";
+    } else {
+        rhs << print_name(name);
+    }
+    rhs << "[" << id_index << "]";
+
+    return rhs.str();
+}
+
 void CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::visit(const Load *op) {
     user_assert(is_one(op->predicate)) << "Predicated load is not supported inside OpenCL kernel.\n";
 
     // If we're loading a contiguous ramp into a vector, use vload instead.
     Expr ramp_base = strided_ramp_base(op->index);
     if (ramp_base.defined()) {
         internal_assert(op->type.is_vector());
-        string id_ramp_base = print_expr(ramp_base);
 
         ostringstream rhs;
-        rhs << "vload" << op->type.lanes()
-            << "(0, (" << get_memory_space(op->name) << " "
-            << print_type(op->type.element_of()) << "*)"
-            << print_name(op->name) << " + " << id_ramp_base << ")";
+        if((op->alignment.modulus % op->type.lanes() == 0) && (op->alignment.modulus % op->type.lanes() == 0)) {
```

Comment:
Nit: this might be easier to read with a line break after the &&

Reply:
I think this is intended to be (modulus, remainder), not (modulus, modulus).1

Reply:
> Nit: this might be easier to read with a line break after the &&

Fixed.

> I think this is intended to be (modulus, remainder), not (modulus, modulus).

Ah, that's embarrassing. Thanks for catching this!